component/component/main.jsonnet

```diff
@@ -26,18 +30,20 @@ local secrets = [
 
 {
   assert params.secrets != null : 'secrets must be set.',
-  assert params.secrets.exoscale != null : 'secrets.exoscale must be set.',
-  assert params.secrets.exoscale.stringData != null : 'secrets.exoscale.stringData must be set.',
-  assert params.secrets.exoscale.stringData.api_key != null : 'secrets.exoscale.stringData.api_key must be set.',
-  assert params.secrets.exoscale.stringData.api_secret != null : 'secrets.exoscale.stringData.api_secret must be set.',
+  assert params.secrets.cronjob != null : 'secrets.cronjob must be set.',
+  assert params.secrets.cronjob.stringData != null : 'secrets.cronjob.stringData must be set.',
+  assert params.secrets.cronjob.stringData.api_key != null : 'secrets.cronjob.stringData.api_key must be set.',
+  assert params.secrets.cronjob.stringData.api_secret != null : 'secrets.cronjob.stringData.api_secret must be set.',
+  assert params.secrets.cronjob.stringData.cluster_server != null : 'secrets.cronjob.stringData.cluster_server must be set.',
+  assert params.secrets.cronjob.stringData.cluster_token != null : 'secrets.cronjob.stringData.cluster_token must be set.',
```

Comment:
This seems to become more boilerplate than value...
```suggestion
```

Reply:
I would rather make it fail during compile time then at runtime. Do you have another suggestion instead of assert?

Reply:
Not really, I'm not too much familiar with jsonnet. no hard feeling here.